bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryDirectDataWriterHelper.java

```diff
@@ -253,6 +259,13 @@ public long finalizeStream() throws IOException {
     if (this.protoRows.getSerializedRowsCount() != 0) {
       sendAppendRowsRequest();
     }
+    try {
+      ApiFutures.allAsList(appendRowsFutures).get();
```

Comment:
There's no validation on the result. What happens if one of the validateAppendRowsResponse() calls throws exception? It is done in another thread

Reply:
It is done on line 206-211. These are validated futures.